### PR TITLE
Add ESP32 pin settings 

### LINKED ISSUE #324: ESP32 support request
Hello,

I want to use USB Host Shield with ESP32(Dev Module), but found this library not support the module.

I personally tried to add ESP32 pin setting to the code, based on ESP8266 related commits.
Although compiling succeeded, it didn't work(`Usb.Init()` failed).

Is there any plan to officially support running USB Host Shield with ESP32 board?

The below are pin settings in my attempt, for your reference.

| USB Host Shield | ESP32 Dev Module | role |
|-----------------|------------------|----------|
| 5V | 5V |  |
| GND | GND |  |
| RESET | EN |  |
| D9 | GPIO17 | INT |
| D10 | GPIO5 | SPI SS |
| D11 | GPIO23 | SPI MOSI |
| D12 | GPIO19 | SPI MISO |
| D13 | GPIO18 | SPI SCK |

### PR DESCRIPTION
**Pull request by kn1cht**
fix #324 
Supported Espressif ESP32.

### CODE REVIEW

**Comment by Lauszus** [comment]
Looks like you need to explicitly cast to uint8_t in the min function, as was the case with the RedBearLab: https://github.com/felis/USB_Host_Shield_2.0/pull/258/files.

Please see the Travis log: https://travis-ci.org/felis/USB_Host_Shield_2.0/jobs/328511356.

**Comment by kn1cht** [comment]
Hmm, so should I add code to define `MFK_CASTUINT8T` in setting.h for ESP8266 and ESP32 ?

https://github.com/felis/USB_Host_Shield_2.0/blob/e7b6c1ca22bcfeada200777a14d14e716ce71a3a/settings.h#L149-L154

**Comment by Lauszus** [comment]
@kn1cht yes please!

**Comment by Lauszus** [comment]
@kn1cht I have fixed the casting warning in master, as I was fixing some other stuff as well. I will just wait for Travis to finished building and then I will merge your code.

File: controllerEnums.h [hunk 1]
```diff
@@ -18,6 +18,10 @@
 #ifndef _controllerenums_h
 #define _controllerenums_h
 
+#if defined(ESP32)
+#undef PS
+#endif
```
Comment (Lauszus) [inline controllerEnums.h]: What is the PS define used for on ESP32?

**Comment by kn1cht** [comment]
@Lauszus Thank you!

**Comment by Lauszus** [comment]
@kn1cht I just merged the code. Thank you for your contribution!

**Pull request merged**